### PR TITLE
XIVY-9572 Move to GitHub Actions

### PR DESCRIPTION
**Pull request by alexsuter**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: build doc
+      run: docker run -u 1001 -v ${{ github.workspace }}/doc-factory-doc:/tmp axonivy/build-container:read-the-docs-2 make -C /doc-build html BASEDIR='/tmp'
```
Comment (ivy-rew) [inline .github/workflows/ci.yml:17]: seeing these advanced pipelines with many steps in common: shouldn't we generalize the common parts in a shared .yaml file and extend it?

Comment (alexsuter) [reply .github/workflows/ci.yml:17]: not possible. a reusable pipeline is a job. jobs are independent from each other.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,53 +0,0 @@
...
-          def networkName = "build-" + random
-          def seleniumName = "selenium-" + random
-          def ivyName = "ivy-" + random
-          sh "docker network create ${networkName}"
-          try {
-            docker.image("selenium/standalone-firefox:4").withRun("-e START_XVFB=false --shm-size=2g --name ${seleniumName} --network ${networkName}") {
-              docker.build('maven', ".").inside("--name ${ivyName} --network ${networkName}") {
-                def phase = env.BRANCH_NAME == 'master' ? 'deploy -DaltDeploymentRepository=nexus.axonivy.com::https://nexus.axonivy.com/repository/maven-releases/ -DaltSnapshotDeploymentRepository=nexus.axonivy.com::https://nexus.axonivy.com/repository/maven-snapshots/' : 'verify'
-                maven cmd: "clean ${phase} -Divy.engine.version=[9.2.0,] -Dmaven.test.failure.ignore=true -Divy.compiler.warnings=false -Divy.engine.list.url=${params.engineListUrl} -Dtest.engine.url=http://${ivyName}:8080 -Dselenide.remote=http://${seleniumName}:4444/wd/hub"
-              }
```
Comment (ivy-rew) [inline Jenkinsfile:39]: all this stuff is no longer required as the action node already has a XVFB and the like? :cool:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alexsuter) [reply .github/workflows/ci.yml:17]: or is there a feature which I'm not aware of?

**Review by ivy-rew — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (alexsuter) [reply Jenkinsfile:39]: selenium just works without any pain here. I'm wondered too.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ivy-rew) [reply .github/workflows/ci.yml:17]: no I didn't see it anywhere. Just thought we could also extend from local pipelines ... as we do it in market-products.

**Pull request merged**